libsql-sqlite3/src/wal.c

```diff
@@ -1999,7 +2001,9 @@ static int walCheckpoint(
   int (*xBusy)(void*),            /* Function to call when busy */
   void *pBusyArg,                 /* Context argument for xBusyHandler */
   int sync_flags,                 /* Flags for OsSync() (or 0) */
-  u8 *zBuf                        /* Temporary buffer to use */
+  u8 *zBuf,                       /* Temporary buffer to use */
+  void *pCbData,                  /* User data passed to xCb */
+  int (*xCb)(void* pCbData, int mxSafeFrame, const unsigned char* pPage, int nPage, int pageNo, int frameNo) /* Checkpoint callback */
```

Comment:
It also needs a detailed description in the comment for this function, so that users understand what this callback is for and when it's called